chore: Collect select set env vars in telemetry

WillDaSilva opened the pull request:
Diff between `environment_context` version `1-1-0` and `1-2-0`:

```diff
*** src/meltano/core/tracking/iglu-client-embedded/schemas/com.meltano/environment_context/jsonschema/1-1-0     2023-01-16 15:30:55.005363075 -0500
--- src/meltano/core/tracking/iglu-client-embedded/schemas/com.meltano/environment_context/jsonschema/1-2-0     2023-01-17 21:57:07.315052616 -0500
***************
*** 6,10 ****
          "name": "environment_context",
          "format": "jsonschema",
!         "version": "1-1-0"
      },
      "type": "object",
--- 6,10 ----
          "name": "environment_context",
          "format": "jsonschema",
!         "version": "1-2-0"
      },
      "type": "object",
***************
*** 40,43 ****
--- 40,57 ----
              "type": "boolean"
          },
+         "notable_flag_env_vars": {
+             "description": "Select environment variable flags",
+             "type": "object",
+             "propertyNames": {
+                 "type": "string"
+             },
+             "additionalProperties": {
+                 "description": "Boolean values of select env vars, or null if the truth value of the env var could not be determined",
+                 "type": [
+                     "boolean",
+                     "null"
+                 ]
+             }
+         },
          "python_version": {
              "description": "The version of Python in use",
***************
*** 163,166 ****
--- 177,181 ----
          "is_dev_build",
          "is_ci_environment",
+         "notable_flag_env_vars",
          "python_version",
          "python_implementation",

```

This will let us see if Meltano is being run in a codespace, among other things. The `is_ci_environment` field has been left in for backwards compatibility, but if you'd prefer @pnadolny13 we can remove it in favor of this new more general approach, and bump the version to `2-0-0` instead.

TODO:
- [ ] Publish the new `environment_context` schema to SnowcatCloud.

Closes #7175

WillDaSilva committed:
chore: Collect select set env vars in telemetry

Closes #7175

WillDaSilva committed:
Address schema lint issues

edgarrmondragon commented on src/meltano/core/tracking/contexts/environment.py:
```diff
@@ -44,9 +44,19 @@ def _get_parent_context_uuid_str() -> str | None:
+        for env_var_name in cls.notable_flag_env_vars:
+            env_var_value = os.environ.get(env_var_name, "0")
+            with suppress(ValueError):
+                if strtobool(env_var_value):
+                    yield env_var_name, env_var_value
```
So we're only checking _if_ the value is truthy, and then only send the raw string value to snowplow?

WillDaSilva replied:
f520ad8 makes it so that all *set* env vars listed are collected. Their values can be `true`, `false`, or `null`, depending on if they're truthy, falsey, or something else respectively.

WillDaSilva committed:
Collect all set notable flags

Values are now either true, false, or null, depending on if the value was truthy, falsey, or something else respectively.

Unset notable flags are not reported.

WillDaSilva committed:
Merge branch 'main' into 7175-telemetry-truthy-env-vars

pnadolny13 reviewed (approved):
@WillDaSilva Looks good to me! I agree that I'd prefer to keep is_ci_environment around, at least for now. We can definitely have a 2.0.0 schema change down the line to remove it though.

WillDaSilva committed:
Merge branch 'main' into 7175-telemetry-truthy-env-vars
